File: CS/UsableDetailPageHeightConverter.cs
using System;
using System.Windows.Data;
using System.Windows.Markup;
using System.Globalization;

namespace PageHeightDemo {
    public class UsableDetailPageHeightConverter : MarkupExtension, IMultiValueConverter {

        public object Convert(object[] values, Type targetType, object parameter, CultureInfo culture) {
            double height = (double)values[0];

            for (int i = 1; i < values.Length; i++) {
                height -= (double)values[i];
            }

            return height;
        }

        public object[] ConvertBack(object value, Type[] targetTypes, object parameter,
        CultureInfo culture) {
            throw new NotSupportedException();
        }


        public override object ProvideValue(IServiceProvider serviceProvider) {
            return this;
        }
    }
}

File: VB/UsableDetailPageHeightConverter.vb
Imports System
Imports System.Windows.Data
Imports System.Windows.Markup
Imports System.Globalization

Namespace PageHeightDemo

    Public Class UsableDetailPageHeightConverter
        Inherits MarkupExtension
        Implements IMultiValueConverter

        Public Function Convert(ByVal values As Object(), ByVal targetType As Type, ByVal parameter As Object, ByVal culture As CultureInfo) As Object Implements IMultiValueConverter.Convert
            Dim height As Double = CDbl(values(0))
            For i As Integer = 1 To values.Length - 1
                height -= CDbl(values(i))
            Next

            Return height
        End Function

        Public Function ConvertBack(ByVal value As Object, ByVal targetTypes As Type(), ByVal parameter As Object, ByVal culture As CultureInfo) As Object() Implements IMultiValueConverter.ConvertBack
            Throw New NotSupportedException()
        End Function

        Public Overrides Function ProvideValue(ByVal serviceProvider As IServiceProvider) As Object
            Return Me
        End Function
    End Class
End Namespace
